CMakeLists.txt

```diff
@@ -1,504 +1,2030 @@
+target_link_libraries (ripple_libs INTERFACE NIH::sqlite3)
+file (MAKE_DIRECTORY
+    ${CMAKE_BINARY_DIR}/sqlite3/include
+    ${CMAKE_BINARY_DIR}/sqlite3/lib)
+target_include_directories (sqlite3
+  PRIVATE
+    src/sqlite
+    src/sqlite/sqlite
+  INTERFACE
+    ${CMAKE_BINARY_DIR}/sqlite3/include)
+add_custom_command (TARGET sqlite3 POST_BUILD
+  COMMAND ${CMAKE_COMMAND} -E copy_directory
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/sqlite/sqlite ${CMAKE_BINARY_DIR}/sqlite3/include
+  COMMAND ${CMAKE_COMMAND} -E copy
+    $<TARGET_FILE:sqlite3> ${CMAKE_BINARY_DIR}/sqlite3/lib
+  BYPRODUCTS ${CMAKE_BINARY_DIR}/sqlite3/include/sqlite3.h)
+
+#[===================================================================[
+   NIH dep: soci
+#]===================================================================]
+
+add_library (soci STATIC
+  #[=========================================================[
+     this unity file is our interpretation of
+     what sources comprise soci-lib.
+     (this must be investigated/confirmed whenever updated)
+  #]=========================================================]
+  src/ripple/unity/soci.cpp
+  # this header is here to create dependency on the sqlite post-build step
+  ${CMAKE_BINARY_DIR}/sqlite3/include/sqlite3.h)
+target_compile_definitions (soci PUBLIC SOCI_HAVE_CXX_C11=1)
+target_include_directories (soci
+  PUBLIC
+    src/soci/src
+    src/soci/include
+    #[=========================================================[
+       HACK for ninja..which doesn't properly see
+       the POST_BUILD dependency on sqlite (TODO: diagnose)
+       so we must include the original source dir
+    #]=========================================================]
+    $<$<STREQUAL:${CMAKE_GENERATOR},Ninja>:
+      ${CMAKE_CURRENT_SOURCE_DIR}/src/sqlite
+    >
+  PRIVATE
+    src
+    src/soci/src/core
+    src/soci/include/private)
+target_link_libraries (soci
+  NIH::sqlite3
+  Ripple::boost)
+add_library (NIH::soci ALIAS soci)
+target_link_libraries (ripple_libs INTERFACE NIH::soci)
+
+#[===================================================================[
+   NIH dep: snappy
+#]===================================================================]
+
+add_library (snappy STATIC
+  src/snappy/snappy/snappy.cc
+  src/snappy/snappy/snappy-sinksource.cc
+  src/snappy/snappy/snappy-stubs-internal.cc)
+target_compile_options (snappy
+  PRIVATE
+    $<$<NOT:$<BOOL:${MSVC}>>:
+      -Wno-deprecated
+      -Wno-unused-function
+    >
+    $<$<BOOL:${is_gcc}>:-Wno-nonnull-compare>)
+target_include_directories (snappy
+  PUBLIC
+    src/snappy/snappy
+    src/snappy/config)
+add_library (NIH::snappy ALIAS snappy)
+target_link_libraries (ripple_libs INTERFACE NIH::snappy)
+
+#[===================================================================[
+   NIH dep: rocksdb
+#]===================================================================]
+
+include (CheckCXXCompilerFlag)
+CHECK_CXX_COMPILER_FLAG ("-std=c++11" COMPILER_SUPPORTS_CXX11)
```

Comment:
I think we can safely remove this as rippled only supports the following min compilers as far as I can tell:
1. GCC 5+
2. Clang x+ (3.7ish probably based on features used that I've seen?)
3. MSVC 2017?

In any regard, all of these compilers support cpp11 flag. I think we should guarantee our expectations of min compiler support very early on (I think we do already) -- well before NIH 3rd party lib CMake stuff.

Reply:
I agree..`common` sets the standard to c++14, so I think this is not needed